AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Storage/CascadeDeleteOperation.swift

```diff
@@ -142,8 +146,12 @@ public class CascadeDeleteOperation<M: Model>: AsynchronousOperation {
         
         let modelIds = queriedModels.map { $0.identifier(schema: self.modelSchema).stringValue }
         
-        associatedModels = await self.recurseQueryAssociatedModels(modelSchema: self.modelSchema, ids: modelIds)
+        if self.log.logLevel >= .debug && modelName != MutationEvent.modelName {
```

Comment:
we are checking MutationEvent.modelName here and then later in the code, we're checking `isDeveloperDefined.MutationEvent.rawValue`, what's the point of creating `isDeveloperDefined` enum if the all the values are not used? A computed property called `isDeveloperDefined` should check if the instance property `modelName` is one of the three system model types, like
```
var isDeveloperDefinedModel Bool {
   return !systemModelNames.contains(modelSchema.name)
}
```

Then use it wherever you want to control the logging, ie.
```
if self.log.logLevel >= .debug && isDeveloperDefinedModel {
 // only log when log level is debug level or higher and is a developer defined model
}
```